### ISSUE TITLE
MqttSourceSpec should fail connection when not providing the requested credentials

### ISSUE THREAD

**Issue by raboof** [akka/alpakka#649]
```
A timeout occurred waiting for a future to complete. Queried 51 times, sleeping 100000000 nanoseconds between each query. (MqttSourceSpec.scala:161)
[info]   org.scalatest.concurrent.Futures$FutureConcept$$anon$1:
[info]   at org.scalatest.concurrent.Futures$FutureConcept$class.tryTryAgain$1(Futures.scala:538)
[info]   at org.scalatest.concurrent.Futures$FutureConcept$class.futureValueImpl(Futures.scala:550)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
[info]   at org.scalatest.concurrent.Futures$class.whenReady(Futures.scala:676)
[info]   at akka.stream.alpakka.mqtt.scaladsl.MqttSourceSpec.whenReady(MqttSourceSpec.scala:23)
[info]   at akka.stream.alpakka.mqtt.scaladsl.MqttSourceSpec$$anonfun$1$$anonfun$apply$mcV$sp$5.apply(MqttSourceSpec.scala:161)
```

Nothing really pops up in logging
```

-rw-rw-r-- 1 travis travis 2.5K Dec 14 11:15 ./target/mqtt.log

2017-12-14 11:15:04,873 INFO  [default-akka.actor.default-dispatcher-2] [akka.event.slf4j.Slf4jLogger]  Slf4jLogger started
2017-12-14 11:15:04,890 DEBUG [default-akka.actor.default-dispatcher-2] [akka.event.EventStream]  logger log1-Slf4jLogger started
2017-12-14 11:15:04,898 DEBUG [default-akka.actor.default-dispatcher-2] [akka.event.EventStream]  Default Loggers started
2017-12-14 11:15:05,319 DEBUG [default-akka.actor.default-dispatcher-3] [akka.event.EventStream]  shutting down: StandardOutLogger started
2017-12-14 11:15:05,448 INFO  [MqttSinkSpec-akka.actor.default-dispatcher-3] [akka.event.slf4j.Slf4jLogger]  Slf4jLogger started
2017-12-14 11:15:05,449 DEBUG [MqttSinkSpec-akka.actor.default-dispatcher-7] [akka.event.EventStream]  logger log1-Slf4jLogger started
2017-12-14 11:15:05,450 DEBUG [MqttSinkSpec-akka.actor.default-dispatcher-7] [akka.event.EventStream]  Default Loggers started
2017-12-14 11:15:06,285 DEBUG [MqttSinkSpec-akka.actor.default-dispatcher-7] [akka.event.EventStream]  shutting down: StandardOutLogger started
2017-12-14 11:15:06,344 INFO  [MqttSinkSpec-akka.actor.default-dispatcher-4] [akka.event.slf4j.Slf4jLogger]  Slf4jLogger started
2017-12-14 11:15:06,346 DEBUG [MqttSinkSpec-akka.actor.default-dispatcher-3] [akka.event.EventStream]  logger log1-Slf4jLogger started
2017-12-14 11:15:06,346 DEBUG [MqttSinkSpec-akka.actor.default-dispatcher-3] [akka.event.EventStream]  Default Loggers started
2017-12-14 11:15:12,290 INFO  [MqttFlowSpec-akka.actor.default-dispatcher-7] [akka.event.slf4j.Slf4jLogger]  Slf4jLogger started
2017-12-14 11:15:12,294 DEBUG [MqttFlowSpec-akka.actor.default-dispatcher-7] [akka.event.EventStream]  logger log1-Slf4jLogger started
2017-12-14 11:15:12,294 DEBUG [MqttFlowSpec-akka.actor.default-dispatcher-7] [akka.event.EventStream]  Default Loggers started
2017-12-14 11:15:12,331 DEBUG [MqttFlowSpec-akka.actor.default-dispatcher-3] [akka.event.EventStream]  shutting down: StandardOutLogger started
2017-12-14 11:15:12,424 INFO  [default-akka.actor.default-dispatcher-2] [akka.event.slf4j.Slf4jLogger]  Slf4jLogger started
2017-12-14 11:15:12,425 DEBUG [default-akka.actor.default-dispatcher-5] [akka.event.EventStream]  logger log1-Slf4jLogger started
2017-12-14 11:15:12,426 DEBUG [default-akka.actor.default-dispatcher-5] [akka.event.EventStream]  Default Loggers started
2017-12-14 11:15:13,133 DEBUG [MqttSinkSpec-akka.actor.default-dispatcher-6] [akka.event.EventStream]  shutting down: StandardOutLogger started
```

**Comment by juanjoDiaz** [comment]
Hi @raboof,

How can I replicate this? I've tried multiple times in my local environment but I can't get that timeout.

**Comment by raboof** [comment]
I'm not sure if I've seen it locally either, I guess some detective work is needed here...

**Comment by ennru** [comment]
Hopefully fixed by #932.

**Comment by ennru** [comment]
Looks good for now.
